src/test/logical_query_plan/lqp_utils_test.cpp

```diff
@@ -138,6 +138,30 @@ TEST_F(LQPUtilsTest, VisitLQP) {
   }
 }
 
+TEST_F(LQPUtilsTest, VisitLQPUpwards) {
+  // clang-format off
+  const auto expected_nodes = std::vector<std::shared_ptr<AbstractLQPNode>>{node_a,
+    PredicateNode::make(greater_than_(a_a, 4)), UnionNode::make(UnionMode::Positions),
+    PredicateNode::make(less_than_(a_a, 4)), PredicateNode::make(equals_(a_a, 4))};
+  // clang-format on
+
+  expected_nodes[4]->set_left_input(expected_nodes[3]);
+  expected_nodes[3]->set_left_input(expected_nodes[1]);
+  expected_nodes[3]->set_right_input(expected_nodes[2]);
```

Comment:
Is the UnionNode meant to have two inputs? Then, the indexes are mixed up. The UnionNode has index `2`.

Reply:
Oups. Will add a check for that.